policy/policy.go

```diff
@@ -23,36 +21,35 @@ type Policy struct {
 }
 
 // Download downloads the given policies
-func Download(ctx context.Context, policies []Policy) {
-	policyDir := filepath.Join(".", viper.GetString("policy"))
-	err := os.MkdirAll(policyDir, os.ModePerm)
+func Download(ctx context.Context, path string, policies []Policy) error {
+	err := os.MkdirAll(path, os.ModePerm)
 	if err != nil {
-		log.G(ctx).Warnf("Error creating policy directory %q: %v\n", policyDir, err)
+		return fmt.Errorf("make policy directory: %w", err)
 	}
 
 	cli, err := auth.NewClient()
 	if err != nil {
-		log.G(ctx).Warnf("Error loading auth file: %v\n", err)
+		return fmt.Errorf("new auth client: %w", err)
 	}
 
 	resolver, err := cli.Resolver(ctx)
 	if err != nil {
-		log.G(ctx).Warnf("Error loading resolver: %v\n", err)
 		resolver = docker.NewResolver(docker.ResolverOptions{})
 	}
 
-	fileStore := content.NewFileStore(policyDir)
+	fileStore := content.NewFileStore(path)
 	defer fileStore.Close()
 
 	for _, policy := range policies {
 		repository := getRepositoryFromPolicy(policy)
 
-		log.G(ctx).Infof("Downloading: %s\n", repository)
```

Comment:
why don't we no longer want this log?

Reply:
It might be a bit too opinionated, but I'm usually against libraries logging info lines. Generally I leave it up to the consumer to be the one responsible for logging, otherwise our public API should also take in a logger so we have control around it so log lines don't show up in tests (or even so API consumers can silence the output if they so desire).